### PR TITLE
Small fix for unhandled exceptions during net.connect issue

### PR DESCRIPTION
**Pull request by viocost**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -55,6 +55,12 @@ var TorControl = function TorControl(opts) {
         }
 
         this.connection = net.connect(params);
+        
+        //Handling connection errors
+        this.connection.once('error', function(err){
+            cb(new Error("Error connecting to control port: " + err));
```
Comment (atd-schubert) [inline index.js]: The callback is optional. Maybe you check if `cb` is `undefined` at the beginning and add a no-operation function to it, or you test (like used in the other parts) if `cb` exists before calling it.

```js
if (cb) {
  cb(new Error("Error connecting to control port: " + err));
}
```

**Review by atd-schubert — CHANGES_REQUESTED** [review]
Please test if the callback function is not `undefined` first...

**Comment by viocost** [comment]
Fixed it

**Review by atd-schubert — APPROVED** [review]
Thanks!

**Comment by atd-schubert** [comment]
I try to publish a new version on npm in the next hours

**Comment by viocost** [comment]
Thank you! That's a great library!

**Pull request merged**